datastream/dataset.py

```diff
@@ -89,8 +89,15 @@ def from_dataframe(dataframe: pd.DataFrame) -> Dataset[pd.Series]:
             get_item=lambda df, index: df.iloc[index],
         )
 
-    def __getitem__(self: Dataset[T], index: int) -> T:
-        '''Get an example ``T`` from the ``Dataset[T]``'''
+    def __getitem__(self: Dataset[T], index: Union[int, slice]) -> Union[T, Dataset[T]]:
+        '''Get an example or slice from the ``Dataset[T]``'''
+        if isinstance(index, slice):
+            dataframe = self.dataframe.iloc[index]
+            return Dataset(
+                dataframe=dataframe,
+                length=len(dataframe),
+                get_item=self.get_item
+            )
         return self.get_item(self.dataframe, index)
```

Comment:
This is a small scope but I think it might be preferable to avoid an early return here?
```
else:
    self.get_item(self.dataframe, index)
```

Reply:
Nevermind. It's an early return regardless